docs/docs/reference/metaprogramming/macros.md

```diff
@@ -135,24 +135,20 @@ expressiveness.
 
 ### From `Expr`s to Functions and Back
 
-The `Expr` companion object contains an implicit `AsFunction` conversion that turns a tree
+The `Expr` companion object contains a `reduce` conversion that turns a tree
 describing a function into a function mapping trees to trees.
 ```scala
 object Expr {
   ...
-  implicit class AsFunction[...](...) { ... }
+  def reduce[...](...)(...): ... =
 }
 ```
-This decorator gives `Expr` the `apply` operation of an applicative functor, where `Expr`s
-over function types can be applied to `Expr` arguments. The definition
-of `AsFunction(f).apply(x)` is assumed to be functionally the same as
+The definition of `Expr.reduce(f)(x)` is assumed to be functionally the same as
 `'{($f)($x)}`, however it should optimize this call by returning the
 result of beta-reducing `f(x)` if `f` is a known lambda expression.
-
-The `AsFunction` decorator distributes applications of `Expr` over function
-arrows:
+`Expr.reduce` distributes applications of `Expr` over function arrows:
 ```scala
-AsFunction(_).apply: Expr[S => T] => (Expr[S] => Expr[T])
+Expr.reduce(_).apply: Expr[(T1, ..., Tn) => R] => ((Expr[T1], ..., Expr[Tn]) => Expr[R])
 ```
```

Comment:
Maybe call it `distribute`?  From [wikipedia](https://en.wikipedia.org/wiki/Modal_logic#Axiomatic_systems):

```
K, Distribution Axiom: □(p → q) → (□p → □q).
```

Reply:
`distribite` is not precise, it only captures a generic property in the types. It would not describe what the method actually does.

Reply:
Also `reduce` itself is not a distribution, `reduce(_)`  is the a distribution.

Reply:
I agree with @liufengyun. Reduce is counter intuitive.

Reply:
Do you have another suggestion?

Reply:
The following sentence does not read well, given the literal meaning of beta-reduction:

> The `Expr` companion object contains a `betaReduce` conversion that turns a tree
describing a function into a function mapping trees to trees.